src/main/java/uk/gov/companieshouse/api/strikeoffobjections/interceptor/authorization/UserAuthorizationInterceptor.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.companieshouse.api.strikeoffobjections.interceptor.authorization;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.servlet.HandlerInterceptor;
+import uk.gov.companieshouse.api.strikeoffobjections.common.ApiLogger;
+import uk.gov.companieshouse.api.strikeoffobjections.model.entity.Objection;
+import uk.gov.companieshouse.api.strikeoffobjections.service.impl.ERICHeaderFields;
+import uk.gov.companieshouse.api.strikeoffobjections.service.impl.ERICHeaderParser;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class UserAuthorizationInterceptor implements HandlerInterceptor {
+
+    private final ApiLogger apiLogger;
+    private final ERICHeaderParser ericHeaderParser;
+
+    public UserAuthorizationInterceptor(ApiLogger apiLogger, ERICHeaderParser ericHeaderParser) {
+        this.apiLogger = apiLogger;
+        this.ericHeaderParser = ericHeaderParser;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+
+        final String requestId = request.getHeader(ERICHeaderFields.ERIC_REQUEST_ID);
+        apiLogger.debugContext(requestId, "Checking current user is authorised for objection id");
```

Comment:
Minor, but user is authorised to do something with the objection and not the 'objection id'.

Reply:
i think we need the id in the message to identify what objection they were trying to access. maybe stick it on the end? 'Checking if current user is authorised to access this objection that has an id of '

Reply:
Sounds good.